### PR TITLE
LocalBearTestHelper: Remove pragma: no cover

### PR DESCRIPTION
**Pull request by jayvdb**
Fixes https://github.com/coala/coala/issues/4158 , and implements `check_invalidity` as requested at https://github.com/coala/coala/issues/2197 .

### CODE REVIEW

File: coalib/testing/LocalBearTestHelper.py [hunk 1]
```diff
@@ -117,15 +128,18 @@ def check_results(self,
                               list,
                               msg='The given results are not a list.')
 
+        if not msg:
+            msg = ("The local bear '{}' doesn't yield the right results."
+                   .format(local_bear.__class__.__name__))
+            if check_order:
+                msg += ' Or the order may be wrong.'
```
Comment (Makman2) [inline coalib/testing/LocalBearTestHelper.py]: maybe you should change the message to `The local bear '{}' yields no result although it should` here, if `results == []`? That moves the complexity to this function and avoids the `msg` parameter^^

**Comment by yukiisbored** [comment]
ack https://github.com/coala/coala/pull/4167/commits/ef830d17c89716b232a5f2205e5ed5c13a675d15

**Comment by yukiisbored** [comment]
ack https://github.com/coala/coala/pull/4167/commits/eee6af1846504e1229090a1237353dd89c0ee168

**Comment by yukiisbored** [comment]
ack https://github.com/coala/coala/pull/4167/commits/09a92fc2fa2b6ad7318757e5b5ba35af347fa1d7

**Comment by jayvdb** [comment]
reack a1aabed f226357 e7bb4b3

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**